docs/user-manual/event-streaming.md

```diff
@@ -0,0 +1,298 @@
+
+That's all ! 
+
+Only remains the plugin declaration; don't forget it !
+
+```cpp
+DQM_PLUGIN_DECL( DeviceEventStreamer, "DeviceEventStreamer" );
+```
+
+Here after is the full file:
+
+
+```cpp
+// DeviceEventStreamer.cc
+#include <dqm4hep/EventStreamer.h>
+#include <DeviceEvent.h> // our event definition
+
+using namespace dqm4hep::core;
+
+class DeviceEventStreamer : public EventStreamer {
+public:
+  ~DeviceEventStreamer() {}
+  Event* createEvent() const;
+  StatusCode write(const Event* const pEvent, xdrstream::IODevice* pDevice);
+  StatusCode read(Event* &pEvent, xdrstream::IODevice* pDevice);
+};
+
+//------------------------------------------------------------------------------
+//------------------------------------------------------------------------------
+
+Event* DeviceEventStreamer::createEvent() const {
+  
+  DeviceEvent* devEvt = new DeviceEvent(); 
+  Event* event = new EventBase<DeviceEvent>(devEvt);
+  return event;
+}
+
+//------------------------------------------------------------------------------
+
+StatusCode DeviceEventStreamer::write(const Event* const pEvent, xdrstream::IODevice* pDevice) {
+  
+  const DeviceEvent* devEvt = pEvent->getEvent<DeviceEvent>();
+
+  if(nullptr == devEvt)
+    return STATUS_CODE_INVALID_PARAMETER;
+  
+  // write all the properties of the DQM4hep event definition
+  if( ! XDR_TESTBIT( pEvent->writeBase( pDevice ) , xdrstream::XDR_SUCCESS ) )
+    return STATUS_CODE_FAILURE;
+    
+  // write temperature
+  if( ! XDR_TESTBIT( pDevice->write<float>( & devEvt->m_temperature ),  , xdrstream::XDR_SUCCESS ) )
+    return STATUS_CODE_FAILURE;
+    
+  // write pressure
+  if( ! XDR_TESTBIT( pDevice->write<float>( & devEvt->m_pressure ),  , xdrstream::XDR_SUCCESS ) )
+    return STATUS_CODE_FAILURE;
+  
+  return STATUS_CODE_SUCCESS;
+}
+
+//------------------------------------------------------------------------------
+
+StatusCode DeviceEventStreamer::read(Event*& pEvent, xdrstream::IODevice* pDevice) {
+
+  pEvent = this->createEvent();
+  DeviceEvent* devEvt = pEvent->getEvent<DeviceEvent>();
+  
+  // read all the properties of the DQM4hep event definition
+  if( ! XDR_TESTBIT( pEvent->readBase( pDevice ) , xdrstream::XDR_SUCCESS ) )
+    return STATUS_CODE_FAILURE;
+    
+  // read temperature
+  if( ! XDR_TESTBIT( pDevice->read<float>( & devEvt->m_temperature ),  , xdrstream::XDR_SUCCESS ) )
+    return STATUS_CODE_FAILURE;
+    
+  // read pressure
+  if( ! XDR_TESTBIT( pDevice->read<float>( & devEvt->m_pressure ),  , xdrstream::XDR_SUCCESS ) )
+    return STATUS_CODE_FAILURE;
+  
+  return STATUS_CODE_SUCCESS;
+}
+
+//------------------------------------------------------------------------------
+
+DQM_PLUGIN_DECL( DeviceEventStreamer, "DeviceEventStreamer" );
+```
+
```

Comment:
Maybe put the file somewhere in Github and link to it, or put it in a separate page on the doc in order to diminish the `clutter` of this page?

Reply:
I will not move it from now. I will wait to have many cases like from other sections in the maunal and then think about how to group them.

Reply:
Seems legit ;)